consensus/cryptarchia-sync/src/downloader.rs

```diff
@@ -0,0 +1,133 @@
+use futures::{future::BoxFuture, stream, stream::BoxStream, StreamExt as _};
+use libp2p::{PeerId, Stream as Libp2pStream};
+use libp2p_stream::Control;
+use tokio::sync::mpsc::Sender;
+use tracing::error;
+
+use crate::{
+    behaviour::{BlocksResponse, RequestStream, ResponseKind, TipResponse},
+    errors::ChainSyncError,
+    messages::{DownloadBlocksRequest, DownloadBlocksResponse, GetTipResponse, RequestMessage},
+    packing::unpack_from_reader,
+    utils,
+    utils::{open_stream, send_message},
+    ChainSyncErrorKind,
+};
+
+pub struct DownloadBlocksTask;
+
+impl DownloadBlocksTask {
+    pub async fn send_tip_request(
+        peer_id: PeerId,
+        control: &mut Control,
+        reply_sender: Sender<BoxStream<'static, Result<ResponseKind, ChainSyncError>>>,
```

Comment:
Why `ResponseKind`? If its a tip it should return a TipResponse, why a mixed stream of responses?